xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/passwd.vm

```diff
@@ -51,32 +42,49 @@ document.observe('xwiki:dom:loaded', function() {
           $doc.save("$services.localization.render('platform.core.profile.passwd.passwordChanged')", true)
         #end
       #end
-      #if($request.password)
-        #if($isValidPassword)
-          <span class='box infomessage'>$services.localization.render('platform.core.profile.passwd.success')</span>
-          <a href = "$doc.getURL("view")">$services.localization.render('platform.core.profile.passwd.return')</a>
-        #else
-          <span class='box errormessage'>$errorMessage</span>
-        #end
+      #if($request.password && $isValidPassword)
+        <span class='box infomessage'>$services.localization.render('platform.core.profile.passwd.success')</span>
+        <a href = "$doc.getURL("view")">$services.localization.render('platform.core.profile.passwd.return')</a>
       #end
       #if(!$isValidPassword)
-        #set($discard = $xwiki.jsfx.use('uicomponents/widgets/validation/livevalidation_prototype.js'))
-        #set($discard = $xwiki.ssfx.use('uicomponents/widgets/validation/livevalidation.css', true))
         $services.localization.render('platform.core.profile.passwd.instructions')
-
-      <form class="xform third" action="" method="post" autocomplete="off">
+        #set ($fields = [])
+        #if (!$hasAdmin)
```

Comment:
Don't forget the `$isCurrentUsersProfile` check, since a user, even if admin, should always be asked for the old password if he's changing his own password.

Most importantly: It's handled on the server side so we need to do it on the UI as well, otherwise an admin user will not be able to change his password (i.e. the server side will always ask for the original password that the UI never offered to fill in).

Generally: You must keep in mind that your refactoring must not affect the previous functionality (and/or reduce the previously supported use cases and edge cases) in any way outside the scope of (or unless it's clearly specified/intended in) the current issue you're fixing.

Reply:
Thanks for your observation, @Enygma2002. I'll investigate it next days.